src/adera/Shaders/PlumeShader.cpp

```diff
@@ -73,8 +78,29 @@ void PlumeShader::draw_plume(ActiveEnt e,
         .setPower(rPlumeComp.m_powerLevel)
         .setTransformationMatrix(entRelative)
         .setProjectionMatrix(camera.m_projection)
-        .setNormalMatrix(entRelative.normalMatrix())
-        .draw(rMesh);
+        .setNormalMatrix(entRelative.normalMatrix());
+
+    // Draw back face
+    Renderer::setFaceCullingMode(Renderer::PolygonFacing::Front);
+    rShader.draw(rMesh);
+
+    // Draw front face
+    Renderer::setFaceCullingMode(Renderer::PolygonFacing::Back);
+    rShader.draw(rMesh);
+}
+
+PlumeShader::RenderGroup::DrawAssigner_t PlumeShader::gen_assign_plume(
+        PlumeShader *pShader)
+{
+    return [pShader]
```

Comment:
my data-ownership senses are tingling here.

If you know that it's 100% safe to be passing around raw-pointers like this, please carefully document the function to indicate as such.

Otherwise, this would be a good opportunity to use a std::weak_ptr if pShader can die, or even an std::shared_ptr, if it can't.

Reply:
Shaders are stored in context resources, which shares its lifetime with the GL context. Most of the time™ the shaders will outlive the draw assignment functions, but it's still possible for the shader to be unloaded if something like that is ever implemented.

Similar to a smart pointer, DependRes can be used here, stored in the lambda somehow, where it can also access the "is loaded" functionality. Now that I think of it, DependRes can possibly be replaced with, or rely upon a shared pointer.